sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuShuffleExchangeExecBase.scala

```diff
@@ -300,12 +300,18 @@ object GpuShuffleExchangeExecBase {
                 batch.close()
                 batch = iter.next()
               }
-              partitioned = getParts(batch).asInstanceOf[Array[(ColumnarBatch, Int)]]
-              partitioned.foreach(batches => {
-                metrics(GpuMetric.NUM_OUTPUT_ROWS) += batches._1.numRows()
-              })
-              metrics(GpuMetric.NUM_OUTPUT_BATCHES) += partitioned.length
-              at = 0
+              // Get a non-empty batch or the last batch. So still need to
+              // check if it is empty for the later case.
```

Comment:
nit, it would be good to move this comment to line 297, and word it slightly differently.
```suggestion
              // Make sure we close all batches with 0 rows since we don't want to shuffle them.
```